app/controllers/lectures_controller.rb

```diff
@@ -75,6 +75,13 @@ def join_lecture
     redirect_to @lecture, notice: "You successfully joined the lecture."
   end
 
+  def leave_lecture
+    @lecture.leave_lecture(current_user)
+    @lecture.save
+    current_user.save
+    redirect_to current_lectures_url, notice: "You successfully left the lecture."
+  end
```

Comment:
Is it necessary to save the current_user here?
While trying to figure this out, I noticed that all tests succeed, even if we don't  save the lecture and/or the user. Same is true for join_lecture. 
I think it still works because lecture.participating_students and students.participating_lectures are not actual attributes of the model objects and are therefore not stored in the database tables. So Model.save does nothing here, because no attributes are changed.
I tested it and its true, lecture.join_lecture already commits the transaction:

![image](https://user-images.githubusercontent.com/38725693/70747471-5dee0a00-1d28-11ea-8ec3-20db8fd9c81d.png)

So I think it's safe to drop those 2  .save lines